src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -634,62 +686,21 @@ private static String getProviderUrl(String server, String rootDN) {
         return server;
     }
 
-    public String getManagerPassword() {
-        return Secret.toString(managerPasswordSecret);
-    }
-
-    public Secret getManagerPasswordSecret() {
-        return managerPasswordSecret;
-    }
-
-    public String getLDAPURL() {
-        return toProviderUrl(getServerUrl(), fixNull(rootDN));
-    }
-
-    public String getDisplayNameAttributeName() {
-        return StringUtils.defaultString(displayNameAttributeName, DescriptorImpl.DEFAULT_DISPLAYNAME_ATTRIBUTE_NAME);
-    }
-
-    public String getMailAddressAttributeName() {
-        return StringUtils.defaultString(mailAddressAttributeName, DescriptorImpl.DEFAULT_MAILADDRESS_ATTRIBUTE_NAME);
-    }
-
     /**
      * Creates security components.
      * @return Created {@link SecurityComponents}
      * @throws IllegalStateException Execution error
      */
     @Override @Nonnull
     public SecurityComponents createSecurityComponents() {
-        Binding binding = new Binding();
-        binding.setVariable("instance", this);
-
-        final Jenkins jenkins = Jenkins.getInstance();
-        if (jenkins == null) {
-            throw new IllegalStateException("Jenkins has not been started, or was already shut down");
-        }
-        
-        BeanBuilder builder = new BeanBuilder(jenkins.pluginManager.uberClassLoader);
-        String fileName = "LDAPBindSecurityRealm.groovy";
-        try {
-            File override = new File(jenkins.getRootDir(), fileName);
-            builder.parse(
-                    new AutoCloseInputStream(override.exists() ? new FileInputStream(override) :
-                        getClass().getResourceAsStream(fileName)), binding);
-        } catch (FileNotFoundException e) {
-            throw new IllegalStateException("Failed to load "+fileName, e);
-        }
-        WebApplicationContext appContext = builder.createApplicationContext();
-
-        ldapTemplate = new LdapTemplate(findBean(InitialDirContextFactory.class, appContext));
-
-        if (groupMembershipStrategy != null) {
-            groupMembershipStrategy.setAuthoritiesPopulator(findBean(LdapAuthoritiesPopulator.class, appContext));
+        LDAPAuthenticationManager manager = new LDAPAuthenticationManager();
+        DelegateLDAPUserDetailsService details = new DelegateLDAPUserDetailsService();
```

Comment:
Would it make sense to not route through the delegate if there is one and only one LDAP configuration? That way existing configurations would behave as before and not deal with the delegation